R_scripts/sapflow.Rmd

```diff
@@ -0,0 +1,471 @@
+---
+title: "Sapflow QA / QC"
+author: "Ben Bond-Lamberty and Stephanie Pennington"
+date: "`r Sys.time()`"
+output:
+  html_document:
+    df_print: paged
+    toc: yes
+    toc_depth: '4'
+    toc_float: yes
+    number_sections: true
+---
+
+```{r setup, echo=FALSE, message=FALSE, warning=FALSE}
+# Load packages
+TESTING <- FALSE
+VOLTAGE_RANGE <- c(0.2, 0.8)  # expected 'good' range for probe voltage diff
+
+# Daily cycle settings
+AMPLITUDE_RANGE <- c(-0.01, -0.2)
+
+# In graphs, TRUE is "ok" while FALSE is "error"
+TEST_COLOR_SCALE <- c("TRUE" = "black", "FALSE" = "red")
+
+# This list holds the test results. The name of each list entry is of the form
+# "WARN.<test_name" or "ERROR.<test_name>".
+# Every entry is a tibble/data frame, with the following required columns:
+# * Tree_Code
+# * Port
+# * failures - an integer
+DAILY_TESTS <- list()
+
+library(readr)
+library(lubridate)
+library(dplyr)
+library(tidyr)
+library(ggplot2)
+library(plotly)
+theme_set(theme_minimal())
+
+# Global chunk options
+knitr:: opts_chunk$set(echo = FALSE)
+
+# Function setup
+read_sapflow <- function(filename) {
```

Comment:
Do you want to move this to a support file, as you did with the TEROS read function?

Reply:
Done!